tools/depcheck/pkg/cmd/trace.go

```diff
@@ -0,0 +1,293 @@
+%[1]s trace --root=pkg/one --root=pkg/two
+
+# create a dependency graph and output in "dot" format
+%[1]s trace --root=pkg/one --output=dot
+`
+
+var (
+	validPackages = []string{
+		"/vendor/bitbucket.org/",
+		"/vendor/cloud.google.com",
+		"/vendor/github.com",
+		"/vendor/go4.org",
+		"/vendor/golang.org",
+		"/vendor/google.golang.org",
+		"/vendor/gopkg.in",
+		"/vendor/k8s.io/",
+		"/vendor/vbom.ml",
+	}
+	validDepPrefixes = []string{"github.com/openshift/origin"}
+)
+
+type TraceImportsOpts struct {
+	BaseDir string
+	Roots   []string
+
+	OutputFormat string
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type TraceImportsFlags struct {
+	OutputFormat string
+	Roots        []string
+}
+
+func (o *TraceImportsFlags) ToOptions(out, errout io.Writer) (*TraceImportsOpts, error) {
+	return &TraceImportsOpts{
+		Roots: o.Roots,
+
+		OutputFormat: o.OutputFormat,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdTraceImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &TraceImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:     "trace --root=pkg/one",
+		Short:   "Creates and analyzes a dependency graph from a set of given root packages",
+		Long:    "Creates and analyzes a dependency graph from a set of given root packages",
+		Example: fmt.Sprintf(traceImportsExample, parent),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts, err := flags.ToOptions(out, errout)
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&flags.Roots, "root", flags.Roots, "set of entrypoints for dependency trees used to generate a depedency graph.")
+	cmd.Flags().StringVarP(&flags.OutputFormat, "output", "o", "", "output generated dependency graph in specified format. One of: dot.")
+	return cmd
+}
+
+func (o *TraceImportsOpts) Complete() error {
+	o.Roots = ensureRecursiveRootFormat(o.Roots)
+	return nil
+}
+
+func ensureRecursiveRootFormat(roots []string) []string {
+	parsedRoots := []string{}
+	for _, root := range roots {
+		if strings.HasSuffix(root, "...") {
+			parsedRoots = append(parsedRoots, root)
+			continue
+		}
+
+		slash := ""
+		if string(root[len(root)-1]) != "/" {
+			slash = "/"
+		}
+		root = strings.Join([]string{root, slash, "..."}, "")
+		parsedRoots = append(parsedRoots, root)
+	}
+
+	return parsedRoots
+}
+
+func (o *TraceImportsOpts) Validate() error {
+	if len(o.Roots) == 0 {
+		return errors.New("at least one root package must be provided")
+	}
+	if len(o.OutputFormat) != 0 && o.OutputFormat != "dot" {
+		return fmt.Errorf("invalid output format provided: %s", o.OutputFormat)
+	}
+
+	return nil
+}
+
+type Node struct {
+	id         int
+	uniqueName string
+}
+
+func (n Node) ID() int {
+	return n.id
+}
+
+type PackageImports struct {
```

Comment:
doc what these are for.  I suspect you're parsing json with them.  If that's the case, make tags to make it obvious